### PR TITLE
SizeF class implicit operator added

### PR DESCRIPTION
**Pull request by Aurioch**
Added implicit operator to Monogame.Extended.SizeF which casts Vector2 to SizeF.

### CODE REVIEW

**Comment by craftworkgames** [comment]
In gitter you mentioned that

> Heh. I also noticed that RectangleF doesn't have a constructor which uses either Point or Vector2, so it might be good idea to add those two for the sake of convenience.

Could you add another `RectangleF` unit test to prove that the constructor already works if you pass it a `Vector2` or `Point`?

**Comment by Aurioch** [comment]
Test added and test passes. I hope I've written correct test, though.

File: Source/MonoGame.Extended.Tests/Shapes/RectangleFTests.cs [hunk 1]
```diff
@@ -24,5 +24,14 @@ public void Rectangle_Intersects_Test()
...
         }
+
+        [Test]
+        public void PassVector2AsConstructorParameter_Test()
+        {
+            var rect1 = new RectangleF(new Vector2(0, 0), new SizeF(12.34f, 56.78f));
+            var rect2 = new RectangleF(new Vector2(0, 0), new Vector2(12.34f, 56.78f));
+
+            Assert.AreEqual(rect1, rect2);
+        }
```
Comment (craftworkgames) [inline Source/MonoGame.Extended.Tests/Shapes/RectangleFTests.cs]: Looks great :)

**Comment by craftworkgames** [comment]
> Test added and test passes. I hope I've written correct test, though.

The trick to knowing if the test is correct is to think about what would make the test fail. For example, in this case, if the implicit conversion didn't exist on the `SizeF` class would the test fail? You could try this by removing the implementation of the implicit conversion (just throw NotImplementedExteption).

In an ideal world, we'd actually write the test first, make sure it fails, then implement the code to make it pass. This is called Test Driven Development (TDD). Some people take this to the extreme, but in reality it's quite difficult to get into this habit.

**Comment by craftworkgames** [comment]
Anyway, I'm good to pull this if you're ready?

**Comment by Aurioch** [comment]
Thanks for the info. Just tried it as you suggested and test fails where it should fail.

So... I'm done here. ^^

**Comment by craftworkgames** [comment]
Thanks for that. It was fun :smile:

**Pull request merged**